src/network/network_admin.cpp

```diff
@@ -336,8 +335,15 @@ NetworkRecvStatus ServerNetworkAdminSocketHandler::SendCompanyInfo(const Company
 		p->Send_uint8(owner);
 	}
 
+
+	uint8 rating = std::count_if(Company::Iterate().begin(), Company::Iterate().end(), [&](auto company){
+	return c->old_economy[0].performance_history < company->old_economy[0].performance_history;});
+
+	p->Send_uint8 (rating);
```

Comment:
I posted this in an earlier comment, I am still not sure about the answer here. But do we actually want to send the rating itself to the client? As they can fetch the performance themselves, and deduce the rating based on it.

One use-case I can imagine is that you only want to fetch the company-info every N minutes and display that on some website. In that case I think it is a valid case, as you don't want to bother with fetching the economy and doing the calculation yourself.

But that said, the one thing that annoys me slightly about this, is that if you fetch multiple companies over time, you can have two companies with the same rating just because the game-state updated between your requests. The `COMPANY_ECONOMY` packet solves this by always sending all companies at once (and also includes the performance rating).

An alternative could be to add the rating in that packet explicit. Albeit a bit silly, it wouldn't hurt either. But changing that packet in a backwards compatible manner is nearly impossible.

Owh, the choices ... :D

Either way, curious what other developers think. @nielsmh , given you replied in the original issue too, how do you view this?

Reply:
When I wrote the reply in the original issue, I didn't realize that it was about sending the ranking position, and that the information for a client to deduce that was already available to the client.

Seeing this now, I'm very inclined to say close this and the original issue.